src/prediction_market/serializers/prediction_market_serializer.py

```diff
@@ -0,0 +1,47 @@
+from django.db.models import Sum
+from rest_framework.serializers import ModelSerializer, SerializerMethodField
+
+from prediction_market.models import PredictionMarket
+from researchhub.serializers import DynamicModelFieldSerializer
+
+
+class PredictionMarketSerializer(ModelSerializer):
+    class Meta:
+        model = PredictionMarket
+        fields = "__all__"
+
+
+class DynamicPredictionMarketSerializer(DynamicModelFieldSerializer):
+    votes = SerializerMethodField()
+    bets = SerializerMethodField()
+
+    class Meta:
+        model = PredictionMarket
+        fields = ["id", "prediction_type", "end_date", "status", "votes", "bets"]
+
+    def get_votes(self, obj):
+        total_votes = obj.prediction_market_votes.count()
+        votes_for = obj.prediction_market_votes.filter(vote=True).count()
+        votes_against = total_votes - votes_for
+
+        return {"total": total_votes, "yes": votes_for, "no": votes_against}
+
+    def get_bets(self, obj):
+        # sum bets for and against
+        bets_for = (
+            obj.prediction_market_votes.filter(
```

Comment:
This will get slow really quickly. We should instead save this into the DB, probably under the PredictionMarket model and every time someone makes a new prediction vote, then update the stored value for that PredictionMarket.

Reply:
I moved them to DB fields, and setup signals to update the values when new votes are created/updated/soft-deleted — that seemed like the best design pattern.